app/controllers/chat_channel_memberships_controller.rb

```diff
@@ -49,12 +49,19 @@ def create
     redirect_to "/chat_channel_memberships/#{@chat_channel.chat_channel_memberships.where(user_id: current_user).first&.id}/edit"
   end
 
-  def remove_invitation
+  def remove_membership
     @chat_channel = ChatChannel.find(params[:chat_channel_id])
     authorize @chat_channel, :update?
-    flash[:settings_notice] = "Invitation Removed."
-    ChatChannelMembership.where(chat_channel_id: @chat_channel.id, id: params[:invitation_id], status: "pending").first&.destroy
-    redirect_to "/chat_channel_memberships/#{@chat_channel.chat_channel_memberships.where(user_id: current_user).first&.id}/edit"
+    @chat_channel_membership = ChatChannelMembership.find(params[:membership_id])
+    if params[:status] == "pending"
+      @chat_channel_membership.destroy
+      flash[:settings_notice] = "Invitation Removed."
+    else
+      @chat_channel_membership.update(status: "removed_from_channel")
+      @chat_channel_membership.remove_from_index!
```

Comment:
I think we already moved chat channel memberships to ES, so I'm not entirely sure if the `remove_from_index` here is still needed, @mstruve will know more.

Reply:
Yes, I think this is correct.